File: 2_C++(VisualStudio)/2_workspace_MFC_Direct2D/1_SideViewGameSample/src/src/ThornBlockObject.cpp
#include "pch.h"
#include "include/ThornBlockObject.h"
#include "GameLibrary/GameLibrary.h"
using namespace Library;

TileType ThornBlockObject::GetTileType() {

    return TileType::Thorn;
}

void ThornBlockObject::Draw(Renderer& renderer, int x, int y, float offsetY, const Camera& camera) {

    GameLibrary::DrawTriangleTile(renderer, x, y, 2, camera,TILE_SIZE ,ColorF::Gray, 0);
}

File: 99_AI生成ソフトウェア(バイブコーディング)/1_Webアプリ系(VSCode)/3_SideViewGameSample_TypeScript_移植元CPlus(Claude)/SideViewGameSample_移植元CPlus/src/GameLogic/GameObject/ThornBlockObject.ts
import { Renderer } from "../Renderer";
import { Camera } from "../Camera";
import { TileType } from "../TileType";
import { ColorF } from "../ColorF";
import { TileMapObject } from "./TileMapObject";
import { GameLibrary } from "../../GameLibrary/GameLibrary";

/**
 * ThornBlockObject.ts
 * 移植元: include/ThornBlockObject.h / src/ThornBlockObject.cpp
 *
 * トゲ(灰色の三角形、上向き固定=direction 0)。
 * 【移植元の挙動をそのまま維持】
 * Draw() の第4引数(offsetY)は使用されず、代わりに DrawTriangleTile への
 * offsetX には常に定数 2 が渡されている(移植元コードのこの実装をそのまま踏襲)。
 */
export class ThornBlockObject extends TileMapObject {

    public override getTileType(): TileType {
        return TileType.Thorn;
    }

    public override draw(renderer: Renderer, x: number, y: number, offsetY: number, camera: Camera): void {
        GameLibrary.drawTriangleTile(renderer, x, y, 2, camera, TileMapObject.TILE_SIZE, ColorF.Gray, 0);
    }
}
